File: src/main/java/com/temelyan/pomoapp/repository/dataJpa/CrudUserRepository.java
package com.temelyan.pomoapp.repository.dataJpa;

import com.temelyan.pomoapp.model.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.transaction.annotation.Transactional;

import java.time.LocalDateTime;
import java.util.Optional;

public interface CrudUserRepository extends JpaRepository<User, Integer> {
    Optional<User> getByEmail(String email);

    @Override
    Optional<User> findById(Integer id);

    @Override
    @Transactional
    User save(User user);

    Optional<User> findByResetToken(String resetToken);

    @Query("select u from User u " +
            "join fetch u.projects pr " +
            "join fetch pr.tasks t " +
            "join fetch t.pomos p " +
            "where u.id=:user_id and (p.finish between :from_date and :to_date)")
    User getUserByIdCompleteGraphForPomosInDateRange(
            @Param("from_date") LocalDateTime from,
            @Param("to_date") LocalDateTime to,
            @Param("user_id") int userId
    );
}

File: src/main/java/com/temelyan/pomoapp/repository/dataJpa/CrudUserRepository.kt
package com.temelyan.pomoapp.repository.dataJpa

import com.temelyan.pomoapp.model.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.transaction.annotation.Transactional
import java.util.*

interface CrudUserRepository : JpaRepository<User, Int> {
    fun getByEmail(email: String): Optional<User>

    override fun findById(id: Int?): Optional<User>

    @Transactional
    fun save(user: User): User

    fun findByResetToken(resetToken: String): Optional<User>
}
